nmigen_boards/mercury.py

```diff
@@ -0,0 +1,226 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.xilinx_spartan_3_6 import *
+from .dev import *
+
+
+__all__ = ["MercuryPlatform"]
+
+
+class MercuryPlatform(XilinxSpartan3APlatform):
+    device  = "xc3s200a"
+    package = "vq100"
+    speed   = "4"
+
+    resources = [
+        Resource("clk50", 0, Pins("P43", dir="i"),
+                 Attrs(IOSTANDARD="LVCMOS33"), Clock(50e6)),
+
+        Resource("user_btn", 0, Pins("P41", dir="i"),
+                 Attrs(IOSTANDARD="LVTTL")),
+
+        # The serial interface and flash memory have a shared SPI bus.
+        # FPGA is secondary.
+        Resource("spiserial", 0,
+            Subsignal("cs", PinsN("P39", dir="i")),
+            Subsignal("clk", Pins("P53", dir="i")),
+            Subsignal("mosi", Pins("P46", dir="i")),
+            Subsignal("miso", Pins("P51", dir="o")),
+            Attrs(IOSTANDARD="LVTTL")
+        ),
+
+        # FPGA is primary.
+        *SPIFlashResources(0,
+            cs="P27", clk="P53", mosi="P46", miso="P51",
+            attrs=Attrs(IO_STANDARD="LVTTL")
+        ),
+
+        # ADC over SPI- FPGA is primary.
+        Resource("adc", 0,
```

Comment:
This should be SPIResources, I think; there's currently no way to override the name, but probably there should be!

Reply:
I agree, and I was going to do this until I saw there was no name override.

Looks like you already noticed this, but just to reiterate, `SPIResources` does _not_ work for `spiserial` because all the SPI pins are reversed by nature of FPGA being the secondary. Perhaps add something to account for this too?

Reply:
We could add a `dir` switch to SPIResources itself, defaulting to `"o"`.